parser/src/python.lalrpop

```diff
@@ -26,32 +26,42 @@ pub Top: ast::Mod = {
 };
 
 Program: ast::Suite = {
-    <lines:FileLine*> => {
-        lines.into_iter().flatten().collect()
-    },
-};
+    => vec![],
+    Statements,
 
-// A file line either has a declaration, or an empty newline:
-FileLine: ast::Suite = {
-    Statement,
-    "\n" => vec![],
 };
 
 Suite: ast::Suite = {
-    SimpleStatement,
-    "\n" Indent <s:Statement+> Dedent => s.into_iter().flatten().collect(),
+    <mut statements:(<SmallStatement> ";")*> <last:SmallStatement> ";"? "\n" => {
+        statements.push(last);
+        statements
+    },
+    "\n" Indent <s:Statements> Dedent => s,
 };
 
-Statement: ast::Suite = {
-    SimpleStatement,
+
+Statements: Vec<ast::Stmt> = {
+    // First simple statement
+    <mut head:(<SmallStatement> ";")*> <last:SmallStatement> ";"? "\n" => {
+        head.push(last);
+        head
+    },
+
+    // The first compound statement
     <s:CompoundStatement> => vec![s],
```

Comment:
The old grammar created a `vec` for every statement. The new grammar simply appends statements at the end of the `vec`.